### PR TITLE
HDFS-15620. RBF: Fix test failures after HADOOP-17281

### PR DESCRIPTION
**Pull request by aajisaka**
JIRA: https://issues.apache.org/jira/browse/HDFS-15620

* Add assertj to test dependency
* Skip testSimpleRootListing in TestRouterWebHDFSContractRootDirectory

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  28m 56s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |   |   0m  0s | [test4tests](test4tests) |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  29m 29s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 40s |  |  trunk passed with JDK Ubuntu-11.0.8+10-post-Ubuntu-0ubuntu118.04.1  |
| +1 :green_heart: |  compile  |   0m 37s |  |  trunk passed with JDK Private Build-1.8.0_265-8u265-b01-0ubuntu2~18.04-b01  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 41s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  15m 23s |  |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 39s |  |  trunk passed with JDK Ubuntu-11.0.8+10-post-Ubuntu-0ubuntu118.04.1  |
| +1 :green_heart: |  javadoc  |   0m 54s |  |  trunk passed with JDK Private Build-1.8.0_265-8u265-b01-0ubuntu2~18.04-b01  |
| +0 :ok: |  spotbugs  |   1m 15s |  |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 13s |  |  trunk passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 32s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 32s |  |  the patch passed with JDK Ubuntu-11.0.8+10-post-Ubuntu-0ubuntu118.04.1  |
| +1 :green_heart: |  javac  |   0m 32s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 29s |  |  the patch passed with JDK Private Build-1.8.0_265-8u265-b01-0ubuntu2~18.04-b01  |
| +1 :green_heart: |  javac  |   0m 29s |  |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 17s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 31s |  |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedclient  |  13m 59s |  |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  |  the patch passed with JDK Ubuntu-11.0.8+10-post-Ubuntu-0ubuntu118.04.1  |
| +1 :green_heart: |  javadoc  |   0m 53s |  |  the patch passed with JDK Private Build-1.8.0_265-8u265-b01-0ubuntu2~18.04-b01  |
| +1 :green_heart: |  findbugs  |   1m 13s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   8m 21s |  |  hadoop-hdfs-rbf in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 109m 56s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2375/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2375 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient xml findbugs checkstyle |
| uname | Linux 9ff351c4f1ec 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 518a212cfff |
| Default Java | Private Build-1.8.0_265-8u265-b01-0ubuntu2~18.04-b01 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.8+10-post-Ubuntu-0ubuntu118.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_265-8u265-b01-0ubuntu2~18.04-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2375/1/testReport/ |
| Max. process+thread count | 3144 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project/hadoop-hdfs-rbf |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2375/1/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs-rbf/pom.xml [hunk 1]
```diff
@@ -114,6 +114,11 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (steveloughran) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/pom.xml:117]: assertj is already exported in test scope for hadoop-common. I'd have expected it to be picked up.

File: hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/fs/contract/router/web/TestRouterWebHDFSContractRootDirectory.java [hunk 2]
```diff
@@ -71,4 +71,9 @@ public void testRmRootRecursive() {
   public void testRmEmptyRootDirRecursive() {
     // It doesn't apply because we still have the mount points here
   }
+
+  @Override
+  public void testSimpleRootListing() {
+    // It doesn't apply because DFSRouter dosn't support LISTSTATUS_BATCH.
```
Comment (steveloughran) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/fs/contract/router/web/TestRouterWebHDFSContractRootDirectory.java:77]: prefer an `Assume.assumeTrue(unsupported, false)` so it goes into the skipped stats. But, given the rest of the tests are the same, not much of an issue.

**Review by steveloughran — APPROVED** [review]

**Review by ayushtkn — APPROVED** [review]
Thanx @aajisaka for the fix.
Would have been good if we would have used `Assume.assumeTrue(unsupported, false)` as Steve suggested.
But none in the class are doing this way. So, not a thing to block this.
Changes LGTM

**Comment by aajisaka** [comment]
Merged. Thank you @steveloughran and @ayushtkn

**Comment by steveloughran** [comment]
@ayushtkn 

> Would have been good if we would have used Assume.assumeTrue(unsupported, false) as Steve suggested.
But none in the class are doing this way. So, not a thing to block this.

All those contract tests are really designed to be driven by an per-FS XML declaration of features. This PR highlights that we may be short of some. Many filesystems don't bother with the root contract, especially if they care about not deleting things...maybe we should revisit that/split it up/add some more configs.

**Pull request merged**